storybook/src/stories/Forms/VpnServerForm.stories.js

```diff
@@ -0,0 +1,424 @@
+      type: { required: true }, // required prop or not
+      control: "none",
+    },
+    "vpnServerRouteProps.onSubmit": {
+      description:
+        "A function that defines what occurs when a modal is submitted",
+      type: { required: true },
+    },
+    "vpnServerRouteProps.onSave": {
+      description: "A function that defines what occurs when the form is saved",
+      type: { required: true },
+    },
+    "vpnServerRouteProps.onDelete": {
+      description:
+        "A function that defines what occurs when the resource is deleted",
+      type: { required: true },
+    },
+    "vpnServerRouteProps.disableSave": {
+      description:
+        "A function that returns a single boolean describing whether the save button should be disabled",
+      type: { required: true },
+      control: "none",
+    },
+    invalidCallback: {
+      description: "Function that determines invalid state for `name` field",
+      type: { required: true }, // required prop or not
+      control: "none",
+    },
+    invalidTextCallback: {
+      description: "Function that determines invalid text for `name` field",
+      type: { required: true }, // required prop or not
+      control: "none",
+    },
+    resourceGroups: {
+      description:
+        "An array of strings containing the names of resource groups to select",
+      type: { required: true }, // required prop or not
+      control: "none",
+    },
+    vpcList: {
+      description: "An array of strings containing the names of VPCs to select",
+      type: { required: true }, // required prop or not
+      control: "none",
+    },
+    subnetList: {
+      description:
+        "An array of strings containing the names of subnet names to choose from",
+      type: { required: true }, // required prop or not
+      control: "none",
+    },
+  },
+  parameters: {
+    docs: {
+      description: {
+        component:
+          "VpnServerForm is a form component that provides functionality for adding or editing a VPN Server.",
+      },
+      inlineStories: false,
+      iframeHeight: 375,
+    },
+  },
+};
+const VpnServerFormStory = () => {
+  function validName(str) {
+    const regex = /^[A-z]([a-z0-9-]*[a-z0-9])?$/i;
+    if (str) return str.match(regex) !== null;
+    else return false;
+  }
+
+  function invalidCallback(stateData) {
+    return (
+      !validName(stateData.name) || contains(["foo", "bar"], stateData.name)
+    );
+  }
+
+  function invalidTextCallback(stateData) {
+    return contains(["foo", "bar"], stateData.name)
+      ? `Name ${stateData.name} already in use.`
+      : `Invalid Name. Must match the regular expression: /^[A-z]([a-z0-9-]*[a-z0-9])?$/i`;
+  }
+
+  function invalidClientIpPoolCallback(stateData) {
+    // functionality to check for overlap will go here...
+    return true;
+  }
+
+  function invalidClientIpPoolTextCallback(stateData) {
+    return `Invalid Client IP Pool. Must be a function that checks for overlap.`;
+  }
```

Comment:
let's change this message just for the sake of the story but other than that lgtm